lldb/source/Plugins/Language/Swift/SwiftOptionSet.cpp

```diff
@@ -174,6 +174,14 @@ bool lldb_private::formatters::swift::SwiftOptionSetSummaryProvider::
       if ((case_value & value) == case_value) {
         // hey a case matched!!
         any_match = true;
+
+        // Print single valued sets without enclosing in brackets.
+        if (first_match && case_value == value) {
+          ss.Printf(".%s", val_name.second.AsCString());
+          dest.assign(ss.GetData());
+          return true;
+        }
```

Comment:
This adds formatting for non-option-sets into the SwiftOptionSetSummaryProvider — would just returning `false` (so the non-optionset formatting is done) work, too?

Reply:
If that doesn't work, I think this should be fine, too.

Reply:
It doesn't work, instead the output is `{ rawValue = ... }`

Reply:
Ah you're right: the `return false` shouldn't be here, but in `lldb_private::formatters::swift::SwiftOptionSetSummaryProvider::
    WouldEvenConsiderFormatting()` — here we already decided on the formatter, so returning false will display nothing.

Reply:
> Ah you're right: the return false shouldn't be here, but in lldb_private::formatters::swift::SwiftOptionSetSummaryProvider:: WouldEvenConsiderFormatting() — here we already decided on the formatter, so returning false will display nothing.

did this work?

Reply:
I didn't try changing `WouldEvenConsiderFormatting`, I'll check now.

Reply:
That function takes only the type, and without changing it to also take a value, it's limited to deciding based on type only.

Reply:
That's a good point! Let's just add a comment explaining why we can't opt-out early in `WouldEvenConsiderFormatting()` to the new special case, then.

Reply:
comment added in https://github.com/apple/llvm-project/pull/1728/commits/3812de837c85225b67ed410eb253ffa271e2fc42